channel/api/snippets/WorkspaceProvisioning/Program.cs

```diff
@@ -0,0 +1,174 @@
+using Google.Apis.Auth.OAuth2;
+using Google.Api.Gax;
+using Google.Cloud.Channel.V1;
+using Google.Type;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace CodeLab
+{
+    class Program
+    {
+        /***************** REPLACE WITH YOUR OWN VALUES ********************************/
+        private static readonly String jsonPrivateKeyFile = "path/to/json_key_file.json";
+        private static readonly String resellerAdminUser = "admin@yourresellerdomain.com";
+        private static readonly String accountId = "C012345";
+        private static readonly String customerDomain = "example.com";
+        /*******************************************************************************/
+
+        private static readonly String accountParent = "accounts/" + accountId;
+
+        private static CloudChannelServiceClient cloudChannelServiceClient;
+
+        static void Main(string[] args)
+        {
+            GoogleCredential credential;
+
+            using (var stream = new FileStream(jsonPrivateKeyFile, FileMode.Open, FileAccess.Read))
+            {
+                credential =
+                    GoogleCredential.FromStream(stream)
+                        .CreateScoped(new string[] {"https://www.googleapis.com/auth/apps.order"})
+                        .CreateWithUser(resellerAdminUser);
+            }
+
+            var client =
+                new CloudChannelServiceClientBuilder
+                    {
+                        TokenAccessMethod = ((ITokenAccess) credential).GetAccessTokenForRequestAsync
+                    }
+                    .Build();
+
+            // For the purpose of this codelab, the code lists all offers and picks
+            // the first offer for Google Workspace Business Standard on an Annual
+            // plan. This is needed because offerIds vary from one account to another,
+            // but this is not a recommended model for your production integration
+            PagedEnumerable<ListOffersResponse, Offer> listOffersResponse =
+                client.ListOffers(new ListOffersRequest 
+                {
+                    Parent = accountParent
+                });
+            
+            Offer selectedOffer = new Offer();
+            String sampleOffer = "Google Workspace Business Standard";
+            String samplePlan = "Commitment";
+
+            foreach (Offer offer in listOffersResponse.ToList())
+            {
+                String offerName = offer.Sku.MarketingInfo.DisplayName;
+                String offerPlan = offer.Plan.PaymentPlan.ToString();
+                if (offerName == sampleOffer && offerPlan == samplePlan)
+                {
+                    selectedOffer = offer;
+                    break;
+                }
+            }
```

Comment:
First suggestion would be to not call ToList(), so you don't fetch all offers.
But really, all of this could be written in a more idiomatic way:
```suggestion
            selectedOffer = listOffersResponse.Where(o => o.Sku.MarketingInfo.DisplayName == sampleOffer && o.Plan.PaymentPlan == samplePlan).FirstOrDefault();
```